### PR TITLE
DST-198: add llm selection tool and base transcript

### PR DESCRIPTION
**Pull request by ccheng26**
## Ticket

Resolves [DST-198](https://navalabs.atlassian.net/browse/DST-198): base code for processing transcripts

## Changes

* added prompt for  transcript
* added selection for openhermes, dolphin, gemini, gpt 3.5 and gpt 4

## Context for reviewers
Logged preliminary results of summarization (stuffing) [here](https://navasage.atlassian.net/wiki/spaces/NL/pages/1094156354/E4+Logged+Results)

## Testing

Testing instructions and expected behavior:
1. `pip install -r requirements.txt`
2. Update `.env` with GOOGLE_API_KEY and OPEN_AI_API_KEY
3. Download models openhermes `ollama pull openhermes` and  dolphin LLM using Ollama: `ollama pull dolphin-mistral` (Install Ollama from https://ollama.ai/)
4. `python run.py`
Review results compared to [evaluation chart](https://navasage.atlassian.net/wiki/spaces/NL/pages/1055883556/E4+Existing+Tool+Evaluation)

### CODE REVIEW

File: 04-call-summaries/run.py [hunk 1]
```diff
@@ -0,0 +1,1916 @@
...
+
+02:32:50.550 --> 02:32:55.940
+
+Rebecca Hale: thank you so much, Mr. Joyce. Hope we have a wonderful rest of your day. Good luck!
+
+
+"""
+
+prompt = """
+You are a helpful AI assistant who will summarize this transcript {transcript}, using the following template:
```
[the remark this answers is not in the document]
Comment (ccheng26) [reply 04-call-summaries/run.py:1855]: I tried this with openhermes and dolphin prior to commiting and it did not follow the format outlined.

[the remark this answers is not in the document]
Comment (yoomlam) [reply 04-call-summaries/run.py:1855]: That might be due to recency bias. Try putting the template at the end of the prompt.

File: 04-call-summaries/llm.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    print("LLM settings:", model_name, settings)
+    # To connect via another URL: Ollama(base_url='http://localhost:11434', ...)
+    return Ollama(model=model_name, callbacks=callbacks, **settings).invoke(prompt)
+
+
+def google_gemini_client(
+    model_name="gemini-pro",
+    prompt=None,
+    settings={},
```
Comment (yoomlam) [inline 04-call-summaries/llm.py:40]: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/dangerous-default-value.html
Install pylint plugin for VSCode to detect such lints.
```suggestion
    settings=None,
```
Also see line 23 above.

**Review by yoomlam — APPROVED** [review]
Approve! pending fix for pylint finding

**Pull request merged**